airflow/config_templates/default_airflow.cfg

```diff
@@ -132,14 +132,6 @@ dag_concurrency = 16
 # Are DAGs paused by default at creation
 dags_are_paused_at_creation = True
 
-# When not using pools, tasks are run in the "default pool",
-# whose size is guided by this config element
-non_pooled_task_slot_count = 128
```

Comment:
This overall feels like it makes the code cleaner, but I do wonder if it's useful to keep this config setting (so that people can tweak the behaviour of the Airflow without _having_ to go in to the UI and change behaviours.

(i.e. we could set the default_pool slots to the config value on bootup?) Not sure.

If we do decide to delete these two config values we should add a note to UPDATING.md about it.